io/net/net_data.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Prudhvi Miryala<mprudhvi@linux.vnet.ibm.com>
+#
+
+# check the statistics of interface, test big ping
+# test lro and gro
+
+import time
+import netifaces
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process
+
+
+class NetDataTest(Test):
+    '''
+    check the statistics of interface, test big ping
+    test lro and gro
+    '''
+    def setUp(self):
+        '''
+            To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
+        network_tools = ("iputils", "ethtool", "net-tools", "openssh-clients")
+        for pkg in network_tools:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.skip("%s package is need to test" % pkg)
+        interfaces = netifaces.interfaces()
+        interface = self.params.get("iface")
+        if interface not in interfaces:
+            self.skip("%s interface is not available" % interface)
+
+    def teststatistics(self):
+        '''
+         check statistics of interface
+        '''
+        self.log.info("Statistic incrementer")
+        interface = self.params.get("iface")
+        peer = self.params.get("peerip")
```

Comment:
Hi Praveen,

No need to pass default interface, if u r not specifying any interface test
will be skipped. i checked already.

Thanks and Regards,
Prudhvi.M



From:	Praveen Pandey <notifications@github.com>
To:	avocado-framework-tests/avocado-misc-tests
            <avocado-misc-tests@noreply.github.com>
Cc:	Prudhvi Miryala04/India/IBM@IBMIN, Author
            <author@noreply.github.com>
Date:	08/31/2016 09:32 PM
Subject:	Re: [avocado-framework-tests/avocado-misc-tests] Testcases for
            check network interface statastics, bigping, lro and gro (#12)



In io/net/net_data.py:


> +        network_tools = ("iputils", "ethtool", "net-tools",
"openssh-clients")
> +        for pkg in network_tools:
> +            if not sm.check_installed(pkg) and not sm.install(pkg):
> +                self.skip("%s package is need to test" % pkg)
> +        interfaces = netifaces.interfaces()
> +        interface = self.params.get("iface")
> +        if interface not in interfaces:
> +            self.skip("%s interface is not available" % interface)
> +
> +    def teststatistics(self):
> +        '''
> +         check statistics of interface
> +        '''
> +        self.log.info("Statistic incrementer")
> +        interface = self.params.get("iface")
> +        peer = self.params.get("peerip")



can peerip and iface declared in setUp ?


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.